src/netspresso_trainer/metrics/detection/metric.py

```diff
@@ -188,6 +188,72 @@ def average_precisions_per_class(
     return average_precisions
 
 
+def precisions_per_class(
+    matches: np.ndarray,
+    prediction_confidence: np.ndarray,
+    prediction_class_ids: np.ndarray,
+    true_class_ids: np.ndarray,
+    num_classes: int,
+    eps: float = 1e-16,
+) -> np.ndarray:
+    sorted_indices = np.argsort(-prediction_confidence)
+    matches = matches[sorted_indices]
+    prediction_class_ids = prediction_class_ids[sorted_indices]
+
+    unique_classes, class_counts = np.unique(true_class_ids, return_counts=True)
+
+    precisions = np.full((num_classes, 1), np.nan)
+
+    for class_idx, class_id in enumerate(unique_classes):
+        is_class = prediction_class_ids == class_id
+        total_true = class_counts[class_idx]
+        total_predictions = is_class.sum()
+
+        if total_true == 0:
+            continue
+        if total_predictions == 0:
+            precisions[int(class_id), 0] = 0.
+            continue
+
+        false_positives = (1 - matches[is_class]).sum(0)
+        true_positives = matches[is_class].sum(0)
+        precision = true_positives / (true_positives + false_positives + eps)
+        precisions[int(class_id), 0] = precision[0]
+    return precisions
+
+def recall_per_class(
+    matches: np.ndarray,
+    prediction_confidence: np.ndarray,
+    prediction_class_ids: np.ndarray,
+    true_class_ids: np.ndarray,
+    num_classes: int,
+    eps: float = 1e-16,
+) -> np.ndarray:
+    sorted_indices = np.argsort(-prediction_confidence)
+    matches = matches[sorted_indices]
+    prediction_class_ids = prediction_class_ids[sorted_indices]
+
+    unique_classes, class_counts = np.unique(true_class_ids, return_counts=True)
+
+    recalls = np.full((num_classes, 1), np.nan)
+
+    for class_idx, class_id in enumerate(unique_classes):
+        is_class = prediction_class_ids == class_id
+        total_true = class_counts[class_idx]
+        total_predictions = is_class.sum()
+
+        if total_true == 0:
+            continue
+        if total_predictions == 0:
+            recalls[int(class_id), 0] = 0.
+            continue
+
+        true_positives = matches[is_class].sum(0)
+        recall = true_positives / (total_true + eps)
+        recalls[int(class_id), 0] = recall[0]
```

Comment:
It seems only use recall of 0.5 threshold .
How about return precisions of all threshold and pick only one in `Recall` class?

And I think renaming `Recall` class to `Recall50` may be less confusing.

Reply:
I push additional commit to fix the issue you've mentioned! Thank you for the review!!